crates/bevy_math/src/curves/mod.rs

```diff
@@ -0,0 +1,38 @@
+use thiserror::Error;
+
+mod fixed;
+mod variable;
+mod variable_linear;
+
+pub use fixed::*;
+pub use variable::*;
+pub use variable_linear::*;
+
+pub type CurveCursor = u16;
+
+pub trait Curve {
+    type Output;
+
+    fn duration(&self) -> f32;
+
+    /// Easier to use sampling method that doesn't needs the keyframe cursor,
+    /// but is more expensive in some types of curve, been always `O(n)`.
+    ///
+    /// This means sampling is more expensive to evaluate as the `time` gets bigger;
+    fn sample(&self, time: f32) -> Self::Output;
```

Comment:
This seems like one of the most important methods; I'd be happy to see some more basic documentation targeted towards beginners who have never animated anything before here.

Reply:
Why ideas on what I should write about?

The purpose of this function is to sample a curve at a given time, there isn't more to it.

Reply:
Hmm. Perhaps this is better addressed at the module / book-level actually. I think that capturing enough information for beginners to make this truly useful requires an Animation 101, which doesn't belong here.